### ISSUE TITLE
Discourage usage of "wildcard imports"

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#528]
**Reported by dietmarw on 12 Apr 2011 09:37 UTC**
Currently the `Modelica.Reference` and also the Modelica Specification contains use cases  of import constructs like:
```text/x-modelica
import Modelica.SIunits.*;
```
or 
```text/x-modelica
import Modelica.Math.*;
```

without pointing out the possible danger of this (models are harder to read and can potentially break when the imported library is enhanced at a later point in time).

Therefore I suggest
 1. Update the MSL and examples from the `Modelica.Reference` and the Specifications to not use those imports. 
 2. And at a place were this kind of (legal) import option is described add a note to discourage its use.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/528

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 22 Sep 2011 05:36 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 6 Mar 2012 15:23 UTC**
The guidelines were updated in 37ba5e4372bd46347fccdccaaec600369fe689fe.
A quick grep for wildcard imports gives:
```
$ ack-grep "import .*\.\*"
Utilities/Strings.mo
540:  import Modelica.Utilities.Strings.*;

Utilities/Examples.mo
8:    import Modelica.Utilities.Strings.*;
253:    import Modelica.Utilities.*;

Thermal/HeatTransfer.mo
6:  import Modelica.SIunits.Conversions.*;

Mechanics/MultiBody/Parts.mo
570:    import Modelica.Math.*;

Mechanics/MultiBody/Frames.mo
328:    import Modelica.Math.*;
1587:      import Modelica.Math.*;

Mechanics/MultiBody/Joints.mo
1552:    import Modelica.Math.*;
1917:    import Modelica.Math.*;
7164:        import Modelica.Math.*;
7541:        import Modelica.Math.*;

Media/package.mo
4960:    import Modelica.Media.*;

Blocks/Continuous.mo
1457:    import Modelica.Math.*;
2292:          import Modelica.Math.*;

Math/package.mo
11821:  import Modelica.Utilities.Streams.*;

Fluid/Machines.mo
304:      import Modelica.SIunits.Conversions.NonSIunits.*;

Fluid/Examples/DrumBoiler.mo
9:    import Modelica.SIunits.Conversions.*;
199:      import Modelica.SIunits.Conversions.*;

Fluid/Examples/AST_BatchPlant.mo
910:      import Modelica.SIunits.Conversions.*;
```
which need fixing.

**Comment by modelica-trac-importer** [comment]
**Comment by ahaumer on 6 Mar 2012 16:01 UTC**
partly (for Modelica.Thermal.HeatTransfer) fixed by 1e33f13c8e4dbda5dfce44645cf13a428af385ae

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 6 Mar 2012 16:07 UTC**
Anton, an alternative would have been (since it allows shorter names):
```diff
- import Modelica.SIunits.Conversions.*;
+ import Cv = Modelica.SIunits.Conversions;
...
- Kelvin = from_degC(Celsius);
+ Kelvin = Cv.from_degC(Celsius);
```
But that's down to personal taste anyway :)

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 7 Mar 2012 13:44 UTC**
Finally fixed in 7ce8f7b0b0d7c0a82ad5a5124df126d4d7c5f5ab.
